### ISSUE TITLE
`block-spacing` errors on multi line blocks

### ISSUE THREAD

**Issue by ben-page** [eslint/eslint#8743]
**Tell us about your environment**

* **ESLint Version:** 4.0.0
* **Node Version:** 8.1.2
* **npm Version:** 4.6.1

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

.eslintrc.js
```js
module.exports = {
    'parserOptions': {
        'ecmaVersion': 2017
    },
    'env': {
        'node': true,
        'es6': true
    },
    'root': true,
    'extends': 'eslint:recommended',
    'rules': {
        'block-spacing': 'error',
		
        //turn off other rules to make output clean
        'max-statements-per-line': 'off',
        'no-unused-vars': 'off'
    }
};
```

**What did you do? Please include the actual source code causing the issue.**
The docs describe the `block-spacing` rule as "Disallow or enforce spaces inside of single line blocks". However, the rule is not limited to single line blocks. It is applied multi line blocks as well.

For example, both functions  below trigger the `block-spacing` rule.

```js
'use strict';
function test() {let i = 0;
    return i;
}

function test2() {//comment
    return 0;
}
```

**What did you expect to happen?**
no errors from `block-spacing` rule

**What actually happened? Please include the actual, raw output from ESLint.**
  2:17  error  Requires a space after '{'  block-spacing
  6:18  error  Requires a space after '{'  block-spacing

✖ 2 problems (2 errors, 0 warnings)
  2 errors, 0 warnings potentially fixable with the `--fix` option.

**Comment by platinumazure** [comment]
This is a documentation problem. Instead of saying single-line blocks, it should say between an open block token and the next token on the same line (or close block token and previous token on the same line).
